src/providers/ldap/sdap_utils.c

```diff
@@ -206,6 +206,39 @@ char *sdap_combine_filters(TALLOC_CTX *mem_ctx,
     return sdap_combine_filters_ex(mem_ctx, '&', base_filter, extra_filter);
 }
 
+/* we check the principal and if it contains our realm, then we drop it
+ * for the comparison with sAMAccountName */
+
+char *principal_string_to_samaccountname(TALLOC_CTX *mem_ctx,
+                                         const char *attr_name,
+                                         const char *princ,
+                                         const char *realm)
+{
+    char *p;
+
+    if (attr_name == NULL || princ == NULL || realm == NULL) {
+        return NULL;
+    }
+
+    p = strchr(princ, '@');
+    if (p == NULL) {
+        return NULL;
+    }
+
+    if (strcmp(p + 1,realm) == 0) {
```

Comment:
Hi,

I'm not sure if this was already a topic before, but I think it might be better to use `strcasecmp()` to make sure the sAMAccountName is added to the search string even if the domain/realm given by the user uses a different upper/local case spelling. Since aAMAccountName is specific to AD and AD does treat those items case-insensitive I think it is ok to directly use `strcasecmp()` without checking the related domain property.

bye,
Sumit

Reply:
Hi,
Right, I was not sure about that - fixed now.
Ondrej

Reply:
Thanks